### ISSUE TITLE
cdktf init pipenv fails

### ISSUE THREAD

**Issue by sonfrau** [hashicorp/terraform-cdk#268]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

I'm on Ubuntu 20.04

Terraform version : v0.12.29
Node version : v17.7.0
Yarn version : 1.22.4
Pipenv : version 2020.6.2
Python 3.8.2

### Affected Resource(s)

I was not able to run your simple python example. I didn't have got any problem with the Typescript example.

Any clue would be pretty thankful for avoiding this behaviour.

### Expected Behavior

New stack/template with no errors.

### Actual Behavior
```
 ~/python-cdktf  $ cdktf init --template="python" --local
Note: By supplying '--local' option you have chosen local storage mode for storing the state of your stack.
This means that your Terraform state file will be stored locally on disk in a file 'terraform.tfstate' in the root of your project.

We will now set up the project. Please enter the details for your project.
If you want to exit, press ^C.

Project Name: (default: 'python-cdktf')
Project Description: (default: 'A simple getting started project for cdktf.')
Warning: Your Pipfile requires python_version 3.7, but you are using 3.8.2 (/home/miquel/.local/share/v/p/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
Pipfile.lock not found, creating…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
Updated Pipfile.lock (a65489)!
Installing dependencies from Pipfile.lock (a65489)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 0/0 — 00:00:00
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
Warning: Your Pipfile requires python_version 3.7, but you are using 3.8.2 (/home/miquel/.local/share/v/p/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
Installing cdktf~=0.0.12-pre.d785af4bf2dad8f1a30b34b8e751ece967c7680d…
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/packaging/requirements.py", line 90, in __init__
    req = REQUIREMENT.parseString(requirement_string)
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/pyparsing.py", line 1654, in parseString
    raise exc
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/pyparsing.py", line 1644, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/pyparsing.py", line 1402, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/pyparsing.py", line 3417, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/pyparsing.py", line 1406, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/pyparsing.py", line 3205, in parseImpl
    raise ParseException(instring, loc, self.errmsg, self)
pkg_resources._vendor.pyparsing.ParseException: Expected stringEnd (at char 18), (line:1, col:19)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3103, in __init__
    super(Requirement, self).__init__(requirement_string)
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/packaging/requirements.py", line 92, in __init__
    raise InvalidRequirement(
pkg_resources.extern.packaging.requirements.InvalidRequirement: Invalid requirement, parse error at "'d785af4b'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/requirements.py", line 975, in _parse_name_from_line
    self._requirement = init_requirement(self.line)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/utils.py", line 200, in init_requirement
    req = Requirement.parse(name)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3150, in parse
    req, = parse_requirements(s)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3096, in parse_requirements
    yield Requirement(line)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3105, in __init__
    raise RequirementParseError(str(e))
pkg_resources.RequirementParseError: Invalid requirement, parse error at "'d785af4b'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/miquel/.local/bin/pipenv", line 8, in <module>
    sys.exit(cli())
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/cli/command.py", line 232, in install
    retcode = do_install(
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/core.py", line 2099, in do_install
    pkg_requirement = Requirement.from_line(pkg_line)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/requirements.py", line 2672, in from_line
    parsed_line = Line(line)
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/requirements.py", line 178, in __init__
    self.parse()
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/requirements.py", line 1313, in parse
    self.parse_name()
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/requirements.py", line 1036, in parse_name
    name = self._parse_name_from_line()
  File "/home/miquel/.local/lib/python3.8/site-packages/pipenv/vendor/requirementslib/models/requirements.py", line 977, in _parse_name_from_line
    raise RequirementError(
pipenv.vendor.requirementslib.exceptions.RequirementError: Failed parsing requirement from 'cdktf~=0.0.12-pre.d785af4bf2dad8f1a30b34b8e751ece967c7680d'
cdktf init [OPTIONS]
```

### References

* #0000

**Comment by cmclaughlin** [comment]
For what it's worth, I haven't seen this problem locally.
If you're really anxious to get started, this PR has some expanded instructions for setting up a local environment - maybe that will help: PR #266 

(I'm just an end user getting started with this project too)

**Comment by Mejato** [comment]
Hi @sonfrau 
I'd recommend you read the following article for a better understanding of pipenv and how it relates to specific versions of python https://pipenv-fork.readthedocs.io/en/latest/basics.html#specifying-versions-of-python

You definitely have a  dependency conflict within you environment, more on this can be found here:
https://www.activestate.com/resources/quick-reads/how-to-manage-python-dependencies-with-virtual-environments/

I'm running the following with no issues:

Ubuntu 20.04
Terraform version : v0.12.29
Node version : v12.18.3
Yarn version : 1.22.4
Pipenv : version 2020.6.2
Python 3.8.2

Using an IDE like PyCharm makes setting up your Python interpreter a breeze. File>Settings>Project:[Projectname]>Python Interpreter.

**Comment by skorfmann** [comment]
Fixed by #270
